invites.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"accounts-service/auth"
+	"accounts-service/models"
+	accountsv1 "accounts-service/protorepo/noted/accounts/v1"
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type invitesAPI struct {
+	accountsv1.UnimplementedInvitesAPIServer
+
+	auth         auth.Service
+	logger       *zap.Logger
+	groupService accountsv1.GroupsAPIServer
+	groupRepo    models.GroupsRepository
+	accountRepo  models.AccountsRepository
+	inviteRepo   models.InvitesRepository
+}
+
+var _ accountsv1.InvitesAPIServer = &invitesAPI{}
+
+func (srv *invitesAPI) SendInvite(ctx context.Context, in *accountsv1.SendInviteRequest) (*accountsv1.SendInviteResponse, error) {
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = srv.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: in.GroupId, AccountId: accountId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "sender not in group")
+	}
+
+	_, err = srv.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: in.GroupId, AccountId: in.RecipientAccountId})
+	if err == nil {
+		return nil, status.Error(codes.InvalidArgument, "recipient already in group")
+	}
+
+	invite, err := srv.inviteRepo.Create(ctx, &models.InvitePayload{SenderAccountID: &accountId, RecipientAccountID: &in.RecipientAccountId, GroupID: &in.GroupId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	// TODO: Invite bevahior ? E-mail etc
```

Comment:
True! Once @maxime-dodin implements the EmailAPI we could totally send an email to the person who got invited! Great idea.